src/main/java/frc/robot/subsystems/intake/commands/BasicIntakeCommand.java

```diff
@@ -0,0 +1,46 @@
+package frc.robot.subsystems.intake.commands;
+
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.Constants;
+import frc.robot.subsystems.intake.Intake;
+
+import java.util.function.BooleanSupplier;
+
+/**
+ * intakes balls.
+ */
+public class BasicIntakeCommand extends CommandBase {
+    private final Intake intake;
+    private final BooleanSupplier isAllianceColor;
+    private final double power;
+
+    public BasicIntakeCommand(Intake intake, BooleanSupplier isAllianceColor, double power) {
+        this.intake = intake;
+        this.isAllianceColor = isAllianceColor;
+        this.power = Constants.Intake.POWER;
+        addRequirements(intake);
+    }
+
+    /**
+     * Set intake motor to an already decided default power.
+     */
+    @Override
+    public void initialize() {
+        intake.openSolenoid();
+        if (isAllianceColor.getAsBoolean())
+            intake.setPower(power);

```

Comment:
This should be in the execute as you want to stop when you sense the ball of the other alliance.